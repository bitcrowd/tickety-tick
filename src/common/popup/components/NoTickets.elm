module NoTickets exposing (..)

import Html exposing (..)
import Html.Attributes exposing (..)

import ExternalLink

import Messages exposing (Msg)


view : Html Msg
view =
  div [ class "text-xs-center m-b-2" ] 
    [ h5 [ class "m-t-2 m-b-2" ] [ text "No tickets found on this page." ]
    , h6 [] [ text "Did you select or open any tickets?" ]
    , p []
      [ text "Tickety-Tick currently supports"
      , br [] []
      , text "GitHub, Jira, Pivotal and Trello."
      ]
    , h6 [] [ text "Missing anything or found a bug?" ]
    , p [] [ ExternalLink.view "https://github.com/bitcrowd/tickety-tick/issues" "Report an issue" ]
    ]
