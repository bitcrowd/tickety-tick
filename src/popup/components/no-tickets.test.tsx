import { render } from "@testing-library/react";
import React from "react";

import type { Props } from "./no-tickets";
import NoTickets from "./no-tickets";

describe("no-tickets", () => {
  function subject(overrides: Partial<Props>) {
    const defaults: Props = { errors: [] };
    const props = { ...defaults, ...overrides };
    return render(<NoTickets {...props} />);
  }

  describe("with no errors", () => {
    it("renders a hint", () => {
      expect(subject({ errors: [] }).asFragment()).toMatchSnapshot();
    });
  });

  describe("with errors", () => {
    it("renders an error report", () => {
      const errors = [new Error("An error to test error reporting")];
      expect(subject({ errors }).asFragment()).toMatchSnapshot();
    });
  });
});
