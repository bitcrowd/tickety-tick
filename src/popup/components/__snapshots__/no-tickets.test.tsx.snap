// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-tickets with errors renders an error report 1`] = `
<div
  className="text-center"
>
  <Report
    errors={
      Array [
        [Error: An error to test error reporting],
      ]
    }
  />
</div>
`;

exports[`no-tickets with no errors renders a hint 1`] = `
<div
  className="text-center"
>
  <Hint />
</div>
`;
